qt/place_page_dialog_user.cpp

```diff
@@ -0,0 +1,291 @@
+#include "qt/place_page_dialog_user.hpp"
+
+#include "qt/qt_common/text_dialog.hpp"
+
+#include "map/place_page_info.hpp"
+#include "indexer/validate_and_format_contacts.hpp"
+#include "platform/settings.hpp"
+
+#include <QtWidgets/QDialogButtonBox>
+#include <QtWidgets/QGridLayout>
+#include <QtWidgets/QHBoxLayout>
+#include <QtWidgets/QLabel>
+#include <QtWidgets/QPushButton>
+
+#include <qboxlayout.h>
+#include <qdialog.h>
+#include <qgridlayout.h>
+#include <qnamespace.h>
+#include <sstream>
+#include <string>
+
+const int place_page_description_max_length = 500;
+const int short_description_minimum_width = 390;
+
+namespace {
+  std::string stripSchemeFromURI(std::string const & input) {
+    std::string result = input;
+    if (size(input) > 8 && ("https://" == input.substr(0, 8))) {
+      result = result.substr(8, size(input));
+    } else if (size(input) > 7 && ("http://" == input.substr(0, 7))) {
+      result = result.substr(7, size(input));
+    }
+    return result;
+  }
+}
```

Comment:
Note formatting and the absence of memory allocations.
```suggestion
namespace
{
std::string_view stripSchemeFromURI(std::string_view uri) {
  for (std::string_view prefix : {"https://", "http://"})
  {
    if (strings::StartsWith(uri, prefix))
      return uri.substr(prefix.size());
  }
  return uri;
}
}  // namespace
```

Reply:
done